src/providers/fault-count/fault-count.ts

```diff
@@ -110,10 +111,20 @@ export class FaultCountProvider {
     }
   }
 
-  public getManoeuvreFaultCount = (category: TestCategory, data: object, faultType: CompetencyOutcome): number => {
+  public getManoeuvreFaultCount = (
+    category: TestCategory,
+    data: object | Manoeuvres[],
+    faultType: CompetencyOutcome,
+  ): number => {
     switch (category) {
-      // TODO(ADI2): Replace with actual fault count helper
-      case TestCategory.ADI2: return sumManoeuvreFaults(data, faultType);
+      case TestCategory.ADI2:
+        if (!Array.isArray(data)) {
```

Comment:
This could be changed for a better check once the `object` is changed to a specific type

Reply:
Regardless of type it will still be an instance or array of instances.

Reply:
`typeof` or `instanceof` would be a more appropriate check

Reply:
`typeof` will always return 'object' though.
Could replace with `!(data instanceof Array)` but doesn't improve readability IMO. Thoughts?

Reply:
So if you change the `object` (see prev comment) then you're going to be able to use the `instanceof` for that

Reply:
Manoeuvre is a Type not a Class and can't be used with `instanceof` operator.

Reply:
Okay, leave it as it is then.